canonical/cos-lib
# Test coordinator cleanup

## Issue
The `ClusterRolesConfig` is lacking the functionality to validate itself and relies on the Coordinator __init__ method to assert on incorrect configurations which does not follow coding best practice.

## Solution
A more testable implementation of the `ClusterRolesConfig` class was introduced in `coordinator.py` which requires manual testing of a distributed deployment 

## Context
Blocks the following PRs:
- [ ] [loki-coordinator-k8-operator](https://github.com/canonical/loki-coordinator-k8s-operator/pull/6)
- [ ] [mimir-coordinator-k8-operator](https://github.com/canonical/mimir-coordinator-k8s-operator/pull/72)
- [ ] [tempo-coordinator-k8-operator](https://github.com/canonical/tempo-coordinator-k8s-operator/pull/26)

## Testing Instructions
Deploy via [distributed-mode-coordinator](https://github.com/canonical/loki-coordinator-k8s-operator/pull/2#distributed-mode-coordinator--read--work--backend--s3--grafana--flog) for each HA Add-on coordinator to verify no errors in charm initialisation and relation events. This requires using an updated `ClusterRolesConfig` for each repo.
- [x] loki-coordinator-k8-operator @MichaelThamm 
- [x] mimir-coordinator-k8-operator @MichaelThamm 
- [ ] tempo-coordinator-k8-operator
  * I am getting tempo-workers not starting. Need to investigate
